server/src/test/java/org/apache/druid/server/shard/SingleDimensionShardSpecTest.java

```diff
@@ -142,6 +146,16 @@ public void testPossibleInDomain()
     Assert.assertTrue(shard7.possibleInDomain(domain2));
   }
 
+  @Test
+  public void testSharePartitionSpace()
+  {
+    final SingleDimensionShardSpec shardSpec = makeSpec("start", "end");
+    Assert.assertTrue(shardSpec.sharePartitionSpace(NumberedPartialShardSpec.instance()));
+    Assert.assertTrue(shardSpec.sharePartitionSpace(new HashBasedNumberedPartialShardSpec(null, 0, 1)));
+    Assert.assertTrue(shardSpec.sharePartitionSpace(new SingleDimensionPartialShardSpec("dim", 0, null, null, 1)));
```

Comment:
How do we end up with these mixed shard spec situations, minor compaction? Like I understand why these evaluate to true given how the checks work, just trying to understand how it happens since afaict only dynamic can be appended.

Reply:
Your understanding is correct. I added `HashBasedNumberedPartialShardSpec` and `SingleDimensionPartialShardSpec` in #7547 but they are not in use yet. They will be used once we support append with non-dynamic or non-dynamic with segment locking.